server/serve.js

```diff
@@ -0,0 +1,70 @@
+import * as fs from "fs";
+import { gzipSync } from "zlib";
+import { createHash } from "crypto";
+import { extname } from "path";
+import { promisify } from "util";
+
+const readFile = promisify(fs.readFile);
+const files = {};
+const MIMETypes = {
+  js: "application/javascript; charset=utf-8",
+  json: "application/json; charset=utf-8",
+  css: "text/css; charset=utf-8"
+};
+
+if (process.env.NODE_ENV === "production") {
+  fs.readdirSync("dist").forEach(filename => {
+    const buffer = fs.readFileSync(`dist/${filename}`);
+    const gzipBuffer = gzipSync(buffer);
+    const md5 = createHash("md5")
+      .update(buffer)
+      .digest("hex")
+      .slice(0, 20);
+
+    files[filename] = { buffer, gzipBuffer, md5 };
```

Comment:
Better filter `filename` before generate gzipped buffer in case there is a big image file get into `dist` somehow.

Overall I think it's a pretty bad idea to keep static assets buffer in memory. Compression should be done by a downstream reverse proxy e.g. Nginx.

Reply:
Well this `serve` middleware is not designed for general purpose file serving. But indeed it's strange to be able to accept a `dir` argument.

So perhaps just hard code `dir` to better inform the intention?

The overall idea is kinda inspired by `koa-static-cache`. At least for this kind of small projects, its convenient to let a standalone Node.js server to be able to serve gzipped files.

Reply:
Well this `serve` middleware is not designed for general purpose file serving. But indeed it's strange to be able to accept a `dir` argument.

So perhaps just hard code `dir` to better inform the intention?

The overall idea is kinda inspired by `koa-static-cache`. At least for this kind of small projects, its convenient to let a standalone Node.js server to be able to serve gzipped files.